dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java

```diff
@@ -434,6 +439,104 @@ public void Test_Secret_Destroy_Method() {
         }
     }
 
+    /**
+     * Test computeSecretWarnings.
+     * Given scenario: You create a secret with no value the secret is expected to be required according to the descriptor.
+     * Expected Result: Such situation should generate a warning
+     */
+    @Test
+    public void Test_Secret_Integrity_Check_Expect_Warning()
+            throws DotDataException, DotSecurityException {
+        final AppsAPI api = APILocator.getAppsAPI();
+
+        final String appKey = "any-key";
+
+        final AppDescriptor descriptor = mock(AppDescriptor.class);
+        when(descriptor.isAllowExtraParameters()).thenReturn(false);
+        final Map<String, ParamDescriptor> params = of(
+        "requiredNoDefault",newParam(null, false, Type.STRING, "any", "hint", true), // This is the rule we intend to break
+        "requiredDefault", newParam("default", false, Type.STRING, "any", "hint", true),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "requiredDefault" appears 4 times in this file; the first occurrence is on line 457](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5519512)